src/scent_block.h

```diff
@@ -70,10 +70,11 @@ struct scent_block {
     }
 
     // We should be working entirely within the range, so don't range check here
-    void apply_gas( const tripoint &p ) {
+    void apply_gas( const tripoint &p, int nintensity = 0 ) {
         const point ndx = index( p );
         assignment[ndx.x][ndx.y].mode = SET;
-        assignment[ndx.x][ndx.y].intensity = 0;
+        assignment[ndx.x][ndx.y].intensity -= nintensity;
+        std::max( 0, assignment[ndx.x][ndx.y].intensity );
```

Comment:
```suggestion
        assignment[ndx.x][ndx.y].intensity = std::max( 0, assignment[ndx.x][ndx.y].intensity );
```

You are not assigning the results back.